app/src/main/java/net/gsantner/markor/format/general/CommonTextActions.java

```diff
@@ -223,76 +230,51 @@ public void onClick(DialogInterface dialog, int selectedColor, Integer[] allColo
 
-        int selectionStart = selection[0];
-        int selectionEnd = selection[1];
+        final Editable text = _hlEditor.getText();
 
-        int lineStart = StringUtils.getLineStart(text, selectionStart);
+        final int[] sel = StringUtils.getSelection(_hlEditor);
+        final int lineStart = StringUtils.getLineStart(text, sel[0]);
+        final int lineEnd = StringUtils.getLineEnd(text, sel[1]);
 
-        String tabString = StringUtils.repeatChars(' ', _tabWidth);
+        if ((up && lineStart > 0) || (!up && lineEnd < text.length())) {
 
-        while (lineStart <= selectionEnd) {
+            final CharSequence line = text.subSequence(lineStart, lineEnd);
 
-            if (deIndent) {
-                int textStart = StringUtils.getNextNonWhitespace(text, lineStart, selectionEnd);
-                int spaceCount = textStart - lineStart;
-                int delCount = Math.min(_tabWidth, spaceCount);
-                int delEnd = lineStart + delCount;
-                if (delCount > 0 && delEnd <= text.length()) {
-                    text.delete(lineStart, delEnd);
-                    selectionEnd -= delCount;
-                }
-            } else {
-                text.insert(lineStart, tabString);
-                selectionEnd += _tabWidth;
-            }
+            final int altStart = up ? StringUtils.getLineStart(text, lineStart - 1) : lineEnd + 1;
+            final int altEnd = StringUtils.getLineEnd(text, altStart);
+            final CharSequence altLine = text.subSequence(altStart, altEnd);
 
-            text = _hlEditor.getText();
-            // Get next line
-            lineStart = StringUtils.getLineEnd(text, lineStart, selectionEnd) + 1;
-        }
+            try {
+                // Prevents changes in text from triggering list prefix insert etc
+                _hlEditor.disableHighlighterAutoFormat();
 
-        _hlEditor.setSelection(
-                StringUtils.getIndexFromLineOffset(text, lStart),
-                StringUtils.getIndexFromLineOffset(text, lEnd));
-    }
+                final int[] selStart = StringUtils.getLineOffsetFromIndex(text, sel[0]);
+                final int[] selEnd = StringUtils.getLineOffsetFromIndex(text, sel[1]);
 
-    public void moveLineBy1(boolean up) {
-        selectWholeLine(true);
-        selectWholeLine(false);
-        String lineToMove = _hlEditor.getText().toString().substring(_hlEditor.getSelectionStart(), _hlEditor.getSelectionEnd());
-        if (!lineToMove.endsWith(LINE_SEPARATOR)) {
-            lineToMove += LINE_SEPARATOR;
-        }
-        _hlEditor.simulateKeyPress(KeyEvent.KEYCODE_DEL);
-        _hlEditor.simulateKeyPress(up ? KeyEvent.KEYCODE_DPAD_UP : KeyEvent.KEYCODE_DPAD_DOWN);
-        _hlEditor.simulateKeyPress(KeyEvent.KEYCODE_MOVE_HOME);
-        _hlEditor.getText().insert(_hlEditor.getSelectionStart(), lineToMove);
-    }
+                final String newPair = String.format("%s\n%s", up ? line : altLine, up ? altLine : line);
```

Comment:
just seeing concat of two lines hair. sure no issue with multiple lines selected?